FS.ios.js

```diff
@@ -23,48 +16,138 @@ var convertError = (err) => {
 var NSFileTypeRegular = RNFSManager.NSFileTypeRegular;
 var NSFileTypeDirectory = RNFSManager.NSFileTypeDirectory;
 
+/**
+ * This function converts the relative path to application home specific directory
+ * Ex: Documents/hello.txt -> <DocumentsDirForApp>/hello.txt
+ * Caches/hello.txt -> <CacheDirForApp>/hello.txt
+ * tmp/hello.txt -> <TmpDirForApp>/hello.txt
+ * @param  {String}
+ * @return {String}
+ */
+function processFilePath(filepath) {
+  var found = filepath.indexOf('/');
+  if (found !== 0) {
+    //relative path, check where is it relative to
+    var splitPath = filepath.split('/');
+    if (splitPath[0] === 'Documents') {
+      splitPath[0] = RNFSManager.NSDocumentDirectoryPath;
+      return splitPath.join('/');
+    } else if (splitPath[0] === 'Caches') {
+      splitPath[0] = RNFSManager.NSCachesDirectoryPath;
+      return splitPath.join('/');
+    } else if (splitPath[0] === 'tmp') {
+      splitPath[0] = RNFSManager.NSTmpDirectoryPath;
+      return splitPath.join('/');
+    }
+    //treat it as relative to Documents directory
+    return RNFSManager.NSDocumentDirectoryPath + '/' + filepath;
+  }
+  //absolute path
+  return filepath;
+}
+
 var RNFS = {
 
-  readDir(path, rootDir) {
-    return _readDir(path, rootDir)
-      .catch(convertError);
+  writeFile(filepath, contents) {
+    var processedFilePath = processFilePath(filepath);
+    if (arguments.length === 4) {
+      RNFSManager.writeFile(processedFilePath, contents, arguments[2], arguments[3]);
+    } else {
+      RNFSManager.writeFile(processedFilePath, contents, {}, arguments[2]);
+    }
   },
 
-  stat(filepath) {
-    return _stat(filepath)
-      .then((result) => {
-        return {
-          'ctime': new Date(result.ctime*1000),
-          'mtime': new Date(result.mtime*1000),
-          'size': result.size,
-          'mode': result.mode,
+  readFile(filepath) {
+    var processedFilePath = processFilePath(filepath);
+    if (arguments.length === 3) {
+      var options = arguments[1];
+      var callback = arguments[2];
+      RNFSManager.readFile(processedFilePath, options, callback);
+    } else {
+      var callback = arguments[1];
+      RNFSManager.readFile(processedFilePath, {
+        encoding: 'null',
+        flag: 'r'
+      }, callback);
+    }
+  },
+
+  readdir(filepath, callback) {
+    var processedFilePath = processFilePath(filepath);
+    RNFSManager.readdir(processedFilePath, callback);
+  },
+
+  mkdir(filepath) {
```

Comment:
Same as for `readFile`:

`mkdir(filepath, mode, callback)`

And then also an updated check for the callback.